aptos-node/src/lib.rs

```diff
@@ -301,14 +328,60 @@ where
     start(config, Some(log_file), false)
 }
 
+/// Merges node_config with the config override file
+fn merge_nodeconfig_override(
+    node_config: NodeConfig,
+    config_override: serde_yaml::Value,
+) -> NodeConfig {
+    serde_merge::tmerge::<NodeConfig, serde_yaml::Value, NodeConfig>(node_config, config_override)
+        .map_err(|e| anyhow::anyhow!("Unable to merge default config with override. Error: {}", e))
+        .unwrap()
+}
+
 /// Creates a single node test config, with a few config tweaks to reduce
 /// the overhead of running the node on a local machine.
 fn create_single_node_test_config(
     config_path: Option<PathBuf>,
+    config_override_path: Option<PathBuf>,
     enable_lazy_mode: bool,
 ) -> NodeConfig {
-    // Build a single validator network with a generated config
-    let mut node_config = NodeConfig::get_default_validator_config();
+    let mut node_config = match config_override_path {
+        // If a config override path was provided, merge it with the default config
+        Some(config_override_path) => {
+            let values = match fs::File::open(&config_override_path) {
+                Ok(mut file) => {
+                    let mut contents = String::new();
+                    match file.read_to_string(&mut contents) {
+                        Ok(_) => serde_yaml::from_str::<serde_yaml::Value>(&contents)
+                            .map_err(|e| {
+                                anyhow::anyhow!(
+                                    "Unable to read config override file as YAML {}. Error: {}",
+                                    config_override_path.to_str().unwrap(),
+                                    e
+                                )
+                            })
+                            .unwrap(),
+                        Err(e) => {
+                            panic!(
+                                "Unable to read contents of config override file {}. Error: {}",
+                                config_override_path.to_str().unwrap(),
+                                e
+                            )
+                        },
+                    }
+                },
+                Err(e) => {
```

Comment:
Is panic'ing a good idea here or should we propagate the error upwards?

Reply:
good point. I've changed it to return `anyhow::Result` so we can propagate the errors upwards